### PR TITLE
xIisModule fixes/Adding unit tests (fixes #182, #215)

### LINKED ISSUE #182: No unit tests for xIisModule
Need to create unit tests for this module

### PR DESCRIPTION
**Pull request by mbreakey3**
This is still in progress. As I started creating unit tests for xIisModule, I found a lot of issues with the module itself, so working on updating that as I write the unit tests.

---

This change is [Reviewable](https://reviewable.io/reviews/powershell/xwebadministration/219)

### CODE REVIEW

**Comment by mbreakey3** [comment]
@kwirkykat @TravisEz13 @tysonjhayes  - I made quite a few changes here because there were a lot of problems with this module (I'm sure there's still more issues too). Would you mind reviewing this?

File: DSCResources/MSFT_xIisModule/MSFT_xIisModule.psm1 [hunk 1]
```diff
@@ -171,242 +150,206 @@ function Set-TargetResource
...
     (
-        [ValidateSet('Present','Absent')]
-        [String] $Ensure,
-        
-        [Parameter(Mandatory = $true)]
+        [Parameter(Mandatory)]
         [String] $Path,
 
-        [Parameter(Mandatory = $true)]
+        [Parameter(Mandatory)]
```
Comment (TravisEz13) [inline DSCResources/MSFT_xIisModule/MSFT_xIisModule.psm1]: I find it more readable with the Mandatory = $false

File: DSCResources/MSFT_xIisModule/MSFT_xIisModule.psm1 [hunk 2]
```diff
@@ -171,242 +150,206 @@ function Set-TargetResource
...
 
-    if(-not $SiteName)
+    if (-not $SiteName)
     {
         return 'IIS:\'
     }
     else
     {
-        return Join-Path 'IIS:\sites\' $SiteName
+        return "IIS:\sites\$SiteName"
```
Comment (TravisEz13) [inline DSCResources/MSFT_xIisModule/MSFT_xIisModule.psm1]: We are supposed to use join in stead of evaluation, why was it removed?

**Comment by TravisEz13** [comment]
:-1:

File: DSCResources/MSFT_xIisModule/MSFT_xIisModule.psm1 [hunk 3]
```diff
@@ -273,7 +273,7 @@ function Get-IisSitePath
     }
     else
     {
-        return "IIS:\sites\$SiteName"
+        return (Join-Path -Path 'IIS:\sites\' -ChilePath $SiteName)
```
Comment (TravisEz13) [inline DSCResources/MSFT_xIisModule/MSFT_xIisModule.psm1]: ChildPath is misspelled

**Comment by mbreakey3** [comment]
Closing this for now, since I'm working on PSDesiredStateConfiguration resources.

**Comment by johlju** [comment]
Reopening this and label it as abandoned since this contains integration tests. It can be continued by anyone in the community.

**Comment by gaelcolas** [comment]
I looked into the code and attempted a rebase, but it's been there for a while (2016) so it's quite a bit of work, and without much knowledge of that module I'd probably waste too much time.

I'll close this one off, but it might be worth just taking 'Tests/Unit/MSFT_xIisModule.tests.ps1' from this PR and submit a new one with just that. Will comment the issue accordingly.

**Pull request closed without merging**